### PR TITLE
Creates FAQ structure for Google search results

### PR DESCRIPTION
**Pull request by sbastn**
Related to https://github.com/dnsimple/dnsimple-business/issues/847. This was recommended by our SEO partner.

A structured FAQ page let's google format question and answers in a rich preview:

I've created a Nanoc filter for the following reasons:
1. The content for the structured FAQ stays always up to date. We update the article, and it updates the rich preview. No copy paste.
2. We can support more FAQ pages in the future.

# Verification:

1. Open `/articles/faq-ssl-certificates/` and view source code. You should see a `application/ld+json` script block before the closing `</head>` tag.
2. Browse to https://search.google.com/test/rich-results.
3. Select Code and paste the source code for the FAQ SSL certificates page
4. Run the test and verify the test results. You can preview the results as well.

### CODE REVIEW

**Comment by aeden** [comment]
I'd like @jacegu to review this, so I am removing myself and putting him in my stead.

**Comment by san983** [comment]
@sbastn not sure if I followed up the QA steps correctly but I got this https://search.google.com/test/rich-results?id=vTtRd33V6QNfKjmiVgpv3g

**Comment by sbastn** [comment]
@san983 You should see a `application/ld+json` script block before the closing `</head>` tag on `/articles/faq-ssl-certificates/`. I have updated the verification steps.

File: lib/structured_data.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if @item[:structured_data]
+      doc = Nokogiri::HTML(content)
+      write_faqs(doc, collect_faqs(doc))
+      return doc.to_html
+    end
+    content
+  end
+
+  def write_faqs(doc, faqs)
+    head = doc.at('head')
```
Comment (jacegu) [inline lib/structured_data.rb]: You can move this line after the template generation (as there is no need to have it here and makes the logic easier to follow.

File: lib/structured_data.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        <% faqs.each_with_index do |faq, index| %>
+          {
+            "@type": "Question",
+            "name": "<%= faq[:question] %>",
+            "acceptedAnswer": {
+              "@type": "Answer",
+              "text": "<%= faq[:answer] %>"
+            }
+          }
+          <%= index != faqs.size - 1 ? "," : "" %>
```
Comment (jacegu) [inline lib/structured_data.rb]: I think this is a little bit hard to read, but I don't have a proposal to make this better that I'm happy with...

File: lib/structured_data.rb (hunk 1 above)
Comment (sbastn) [reply lib/structured_data.rb]: ✅

File: lib/structured_data.rb (hunk 2 above)
Comment (sbastn) [reply lib/structured_data.rb]: I am going to extract this in a method. That could help

**Pull request merged**